src/utils/hooks/useNamespacesResources.ts

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react';
+
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type useNamespacedResourcesType = <T extends K8sResourceCommon>(
+  namespace: string,
+  groupVersionKind: { group: string; kind: string; version: string },
+) => [T[], boolean, any];
+
+const useNamespacedResources: useNamespacedResourcesType = <T extends K8sResourceCommon>(
```

Comment:
The file name and component name aren't consistent. I assume you meant for the file name to `useNamespacedResources`, too?